File: Trainings/src/lesson200408/homeWork/Graph.py
class Graph:

    def __init__(self, n):
        self.numberOfNodes = n
        self.graph = []
        for i in range(0,n):
            a = [0]*n
            self.graph.append(a)

    def getNumberOfNodes(self):
        return self.numberOfNodes
    
    def addEdge(self, a ,b):
        self.graph[a][b] = 1
        self.graph[b][a] = 1

    def removeEdge(self, a ,b):
        self.graph[a][b] = 0
        self.graph[b][a] = 0
    
    def getNeighbors(self, n):
        ret = []
        for i in range(0, self.numberOfNodes):
            if self.graph[n][i] == 1:
                ret.append(i)
        return ret

    

File: Trainings/src/lesson200408/homeWork/Graph.java
import java.util.ArrayList;
import java.util.List;

public class Graph {

    private int [][] graph;
    private int numberOfNodes;

    public Graph(int n){
        graph = new int[n][n];
        numberOfNodes = n;
    }

    public int getNumberOfNodes(){
        return numberOfNodes;
    }

    public void addEdge(int a, int b){
        graph[a][b] = 1;
        graph[b][a] = 1;
    }

    public void removeEdge(int a, int b){
        graph[a][b] = 0;
        graph[b][a] = 0;
    }

    public List<Integer> getNeighbors(int node){
        List<Integer> list = new ArrayList<>();
        for(int i = 0; i < numberOfNodes; i++){
            if(graph[node][i] == 1){
                list.add(i);
            }
        }
        return list;
    }
}
